tests/test_bgp_af.rb

```diff
@@ -695,414 +732,79 @@ def test_maximum_paths_ibgp
-      ['192.168.7.0/24', 'rtmap3'],
-    ]
-    sl2 = [
-      ['192.168.55.0/24', 'rtmap55'],
-      ['192.168.6.0/24', 'rtmap2'],
-      ['192.168.7.0/24', 'rtmap3'],
-      ['192.168.9.0/24'],
-    ]
-    # Test ipv4 unicast vrf default
-    asn = '55'
-    vrf = 'default'
-    af = 'ipv4 unicast'
-    networks_delta_is_lessthan_should(asn, vrf, af, il1, sl1, il2, sl2)
-
-    # Lists are identical except for additional item 4 in
-    # should list (sl1)
-    il1 = [
-      ['2000:123:38::/64', 'rtmap1'],
-      ['2000:123:39::/64', 'rtmap2'],
-      ['2000:123:40::/64', 'rtmap3']]
-    sl1 = [
-      ['2000:123:38::/64', 'rtmap1'],
-      ['2000:123:39::/64', 'rtmap2'],
-      ['2000:123:40::/64', 'rtmap3'],
-      ['2000:123:41::/64'],
-    ]
-    # Item 1 in both lists are different
-    # Additonal item 4 in should list (sl2)
-    il2 = [
-      ['2000:123:38::/64', 'rtmap1'],
-      ['2000:123:39::/64', 'rtmap2'],
-      ['2000:123:40::/64', 'rtmap3'],
-    ]
-    sl2 = [
-      ['2000:155:55::/64', 'rtmap1'],
-      ['2000:123:39::/64', 'rtmap2'],
-      ['2000:123:40::/64', 'rtmap3'],
-      ['2000:123:41::/64'],
-    ]
-    # Test ipv6 unicast vrf default
-    asn = '55'
-    vrf = 'default'
-    af = 'ipv6 unicast'
-    networks_delta_is_lessthan_should(asn, vrf, af, il1, sl1, il2, sl2)
-  end
-
-  def test_networks_scale
-    asn = '55'
-    vrf = 'blue'
-    af = %w(ipv4 unicast)
-    bgp_af = RouterBgpAF.new(asn, vrf, af)
-
-    #
-    # Configure 50 networks and then add and remove from the list
-    #
-    should_list = []
-    (1..50).each do |x|
-      bgp_af.network_set("192.168.#{x}.5/24", "rtmap#{x}")
-      should_list.push ["192.168.#{x}.0/24", "rtmap#{x}"]
-    end
-
-    config_list = bgp_af.networks_delta(should_list)
-    assert_empty(config_list[:add],
-                 'Error: config_list[:add] should be empty')
-    assert_empty(config_list[:remove],
-                 'Error: config_list[:remove] should be empty')
-
-    # Apply config_list
-    bgp_af.networks = config_list
-
-    # Verify is_list on device
-    should_list.each do |network|
-      assert_includes(bgp_af.networks, network,
-                      "Error: device should contain network #{network}")
-    end
-
-    # Change the second half of the list to new values
-    (25..50).each { |x| should_list[x] = ["10.168.#{x}.0/24", "rtmap#{x}"] }
-
-    # Add new values with route maps
-    (51..80).each { |x| should_list[x] = ["10.55.#{x}.0/24", "rtmap#{x}"] }
-
-    # Add new values without route maps
-    (81..90).each { |x| should_list[x] = ["10.55.#{x}.0/24"] }
-
-    # Apply config_list
-    bgp_af.networks = bgp_af.networks_delta(should_list)
-
-    # Verify is_list on device
-    should_list.each do |network|
-      assert_includes(bgp_af.networks, network,
-                      "Error: device should contain network #{network}")
+  def network_cmd(af, dbg, afi)
+    # Initial 'should' state
+    if /ipv6/.match(afi)
```

Comment:
Nit: you don't really need to pass afi to the method since you can grep for ipv6 in dbg:
`if dbg[/ipv6/]`

Fine if you want to keep it though.